[ENT-416] Data sharing consent - UI issue on mobile view

replaceafill opened the pull request:
**Description:** This PR fixes the layout issue of the grant data sharing modal when the screen is narrow. The modal content is also scrollable now.

**JIRA:** [ENT-416](https://openedx.atlassian.net/browse/ENT-416)

**Screenshots:**
![narrow](https://user-images.githubusercontent.com/560781/26954724-ea9af144-4c6e-11e7-90ac-1de077a06402.png)
![scroll](https://user-images.githubusercontent.com/560781/26954725-edf3e1e8-4c6e-11e7-8393-21c3bf0b45f0.png)

**Dependencies:** None 

**Merge deadline:** None

**Installation instructions:** Standard enterprise installation.

**Testing instructions:**

1. Set up a devstack with this branch.
2. Add an Enterprise Customer and enable data sharing consent at enrollment.
3. Add a learner to the Enterprise Customer.
4. Log in as the learner to get the grant_data_sharing_permissions view.
5. Click the "No, take me back." link to display the popup.
6. Make the browser window very narrow and verify that its layout doesn't break.
7. To test the scrolling of the modal you can either modify the `enterprise/templates/enterprise/grant_data_sharing_permissions.html` template in the branch or use your browser developer tools to add content to the `.modal-content div.scrollable` element.

**Merge checklist:**

- [ ] ~Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.~
- [X] All reviewers approved
- [X] CI build is green
- [X] Version bumped
- [X] Changelog record added
- [X] Documentation updated (not only docstrings)
- [X] Commits are (reasonably) squashed
- [X] Translations are updated
- [X] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)

<!--
**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
-->

**Reviewers**
- [ ] @mtyaka

edx-webhook commented:
Thanks for the pull request, @replaceafill! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=126&repo=edx%2Fedx-enterprise"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

mtyaka commented on enterprise/static/enterprise/grant_data_sharing_permissions.css:
```diff
@@ -200,6 +198,13 @@ h2#modal-header-text {
 }
 
+.modal-content .scrollable {
+    max-height: 90vh;
```
nit: you could set it to `calc(90vh - 30px)` to account for the height of the top bar with the close (X) button. That way there will always be some vertical space between the window and the popup (currently if you make the height of the window small enough, the popup fills the entire height).

mtyaka commented on enterprise/templates/enterprise/grant_data_sharing_permissions.html:
```diff
@@ -110,13 +110,15 @@ <h2 class="consent-title">{{ consent_message_header }}</h2>
-                    <p>{{ confirmation_alert_prompt_warning }}</p>
-                    <button class="consent-agreement-button" id="modal-no-consent-button">{{ confirmation_modal_affirm_decline_text }}</button>
-                    <button class="failure-link" id="review-policy-link">{{ confirmation_modal_abort_decline_text }}</button>
+                    <div class="scrollable">
```
nit: I suggest a less generic name, so that future unrelated CSS `.scrollable` additions in the platform don't accidentally affect this particular div. Perhaps `consent-confirmation-modal-scrollable`.

mtyaka commented:
I added two nits, but this looks good! :+1:

- [x] I tested this: I followed the testing instructions in the PR description and tested with various window sizes.
- [x] I read through the code
- [x] I checked for accessibility issues
- [ ] ~~Includes documentation~~ not required

itsjeyd commented:
@brittneyexline @douglashall A couple nits to address but this is ready for you to have a look.

chris-mike commented:
My only concern is for 2 scroll bars. When the modal is open, we do not need the "outer" scroll. It would be better if the modal consumed the screen so that only the modal's scroll bar is necessary.

replaceafill commented:
@chris-mike Ah thanks pointing that out. We could use the same approach we used in the TOS modal (https://github.com/edx/edx-platform/pull/15038) and in the course overlay (https://github.com/edx/edx-enterprise/pull/110).

@douglashall I'll work on that.

replaceafill commented:
@mtyaka I just addressed your comments and prevented the two scrollbars as @chris-mike and @douglashall requested.

Could you please test/review again?

I rebased from 039ee38 btw.

replaceafill commented:
New screenshot:

![update](https://user-images.githubusercontent.com/560781/27049714-62510cb8-4f6d-11e7-8ed2-f60af9db377a.png)

mtyaka commented:
Thanks @replaceafill! I retested and verified that the double scrollbars are gone. Code looks good, too. 👍

replaceafill commented:
@douglashall @itsjeyd @mtyaka I just went through the merge checklist. I rebased from 0f51c68

itsjeyd commented:
Thanks @replaceafill and @douglashall!
